healthcheck/api.py

```diff
@@ -102,7 +102,11 @@ def add_watcher(name):
         return "watcher is required", 400
 
     watcher = data["watcher"]
-    get_manager().add_watcher(name, watcher)
+    password = None
+    if "password" in data:
+        password = data["password"]
```

Comment:
You could use `password = data.get("password")`. It already return `None` for key not found in dict